package jetstream;

import "jetstream_types.proto";
import "jetstream_controlplane.proto";

message TableRequest {
  repeated string names = 1;
}

message TableResponse {
  required Cube tables = 1;	
}


message WorkerRequest {
  enum Type {GET_TABLE =1;ALTER=2;}
  required Type type = 1;

  optional TableRequest table_request = 2;
  optional AlterTopo alter = 3;
}


/* An RPC response from a JetStream worker node.
* Currently there is only an optional error message field.
*
* TODO: should probably have IDs for requests so we can have multiple concurrent
* request/responses from a single client
*/
message WorkerResponse {
  optional string error = 147; //use unusual number to catch errors on parsing
}


service TableService {
  rpc getTable (TableRequest) returns (TableResponse);
}
